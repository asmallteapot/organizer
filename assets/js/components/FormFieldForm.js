import React from 'react'
import { Form, Text, Checkbox, Select } from 'react-form'
import _ from 'underscore'

export default class FormFieldForm extends React.Component {
  render() {
    var inputs = [];
    _.each(this.props.fields, (field) => {
      var element;
      switch (field.control_type) {
        case 0: // text
          element = (<Text field={field.id} />);
          break;
        case 1: // boolean
          element = (<Checkbox value='true' field={field.id} />);
          break;
        case 2: // multiple choice
        case 3: // options
          var options = _.map(
            field.control_data.split("\n"),
            (line) => {
              return {label: line, value: line};
            }
          );
          element = (<Select field={field.id} options={options} />);
      }
      inputs.push((
        <label key={field.name}>{field.name} {element}</label>
      ));
    });
    return (
      <Form>
        {inputs}
      </Form>
    )
  }
}
