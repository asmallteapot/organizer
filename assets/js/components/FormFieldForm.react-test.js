import React from 'react'
import FormFieldForm from './FormFieldForm'
import { mount } from 'enzyme'

test('Smoke test', () => {
  const component = mount(<FormFieldForm fields={[]} />);

  let tree = component.html();
  expect(tree).toMatchSnapshot();
});

test('Single text field', () => {
  const fields = [{
    control_type: 0,
    id: 'field',
    name: 'Field Name'
  }]
  const component = mount(<FormFieldForm fields={fields} />);

  let tree = component.html();
  expect(tree).toMatchSnapshot();
  expect(component.find('input')).toHaveLength(1);
});

test('Single checkbox field', () => {
  const fields = [{
    control_type: 1,
    id: 'field',
    name: 'Field Name'
  }]
  const component = mount(<FormFieldForm fields={fields} />);

  let tree = component.html();
  expect(tree).toMatchSnapshot();
  expect(component.find('input[type="checkbox"]')).toHaveLength(1);
  expect(component.find('input')).toHaveLength(1);
});
