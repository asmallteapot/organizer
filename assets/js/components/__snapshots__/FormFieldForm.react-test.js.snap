// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Single checkbox field 1`] = `"<div class=\\"ReactForm\\"><label><!-- react-text: 3 -->Field Name<!-- /react-text --><!-- react-text: 4 --> <!-- /react-text --><div class=\\"FormInput\\"><input type=\\"checkbox\\" value=\\"true\\"><div class=\\"FormError\\" style=\\"display: none;\\"></div></div></label></div>"`;

exports[`Single text field 1`] = `"<div class=\\"ReactForm\\"><label><!-- react-text: 3 -->Field Name<!-- /react-text --><!-- react-text: 4 --> <!-- /react-text --><div class=\\"FormInput\\"><input value=\\"\\"><div class=\\"FormError\\" style=\\"display: none;\\"></div></div></label></div>"`;

exports[`Smoke test 1`] = `"<div class=\\"ReactForm\\"></div>"`;
